### PR TITLE
Ensure that swagger works for multiple versions concurrently.

### PR DESCRIPTION
**Pull request by jessemyers**
The swagger endpoint was fixed, independent of version. This meant we couldn't have multiple versions.

### CODE REVIEW

File: microcosm_flask/namespaces.py [hunk 1]
```diff
@@ -95,30 +98,30 @@ def endpoint_for(self, operation):
...
-                self.subject_name,
-                operation.value.name,
-                self.object_name,
-            )
-        else:
-            return operation.value.pattern.format(
-                self.subject_name,
-                operation.value.name,
-            )
+        return operation.value.pattern.format(
```
Comment (jessemyers) [inline microcosm_flask/namespaces.py]: I switched to keyword-based format strings so I can substitute without having to know which pattern is being used.

File: microcosm_flask/operations.py [hunk 2]
```diff
@@ -27,6 +29,7 @@ class Operation(Enum):
     """
     # discovery operation
     Discover = OperationInfo("discover", "GET", NODE_PATTERN, 200)
+    DiscoverVersion = OperationInfo("discover_version", "GET", VERSIONED_NODE_PATTERN, 200)
```
Comment (jessemyers) [inline microcosm_flask/operations.py]: Swagger now embeds the version into its endpoint pattern so we can have multiple at once with different versions.

File: microcosm_flask/operations.py [hunk 3]
```diff
@@ -55,3 +58,15 @@ def from_name(cls, name):
...
         else:
             raise ValueError(name)
+
+    @property
+    def endpoint_pattern(self):
+        """
+        Convert the operation's pattern into a regex matcher.
+
+        """
+        parts = self.value.pattern.split(".")
```
Comment (jessemyers) [inline microcosm_flask/operations.py]: This builds a matcher so we can extract kwargs back from the endpoint.

File: microcosm_flask/swagger/schema.py [hunk 4]
```diff
@@ -38,6 +38,16 @@
 }
 
 
+SWAGGER_TYPE = "__swagger_type__"
+SWAGGER_FORMAT = "__swagger_format__"
+
+
+def swagger_field(field, swagger_type="string", swagger_format=None):
```
Comment (jessemyers) [inline microcosm_flask/swagger/schema.py]: I added a wrapper function for a marshmallow field so we can more easily attach the customization attributes.

**Comment by adamhadani** [comment]
LGTM

**Pull request closed without merging**